CreeDictionary/API/models.py

```diff
@@ -356,10 +361,68 @@ def fetch_lemma_by_user_query(user_query: str, **kwargs) -> "CreeAndEnglish":
         user_query = user_query.lower()
 
         # build up result_lemmas in 2 ways
-        # 1. spell relax in descriptive fst
+        # 1. affix search (return all results that ends/starts with the query string)
+        # 2. spell relax in descriptive fst
         # 2. definition containment of the query word
+
         cree_results: Set[CreeResult] = set()
 
+        # there will be too many matches for some shorter queries
+        if len(user_query) > settings.AFFIX_SEARCH_THRESHOLD:
+            # prefix search
+            no_diacritics_query_string = remove_cree_diacritics(user_query)
+            possible_wf_ids = Wordform.N_TH_LETTER_TO_LEMMA_IDS[0][
+                no_diacritics_query_string[0]
+            ].copy()
+            for letter_index, letter_to_ids in enumerate(
+                Wordform.N_TH_LETTER_TO_LEMMA_IDS
+            ):
+                if letter_index == 0:
+                    # accounted for before the for loop
+                    continue
```

Comment:
In a past commit I used All_IDS & LETTER_1 & LETTER_2 & LETTER_3 .... The ALL_IDs part is redundant and a waste of performanc.